google-compute-engine/src/main/java/org/jclouds/googlecomputeengine/features/InstanceApi.java

```diff
@@ -253,6 +253,27 @@ Operation setScheduling(@PathParam("instance") String instanceName,
                            @PayloadParam("automaticRestart") boolean automaticRestart);
 
    /**
+    * Sets an instance's scheduling options.
+    * @see <a href = "https://cloud.google.com/compute/docs/instances#onhostmaintenance"/>
+    *
+    * @param instanceName The name of the instance
+    * @param onHostMaintenance either MIGRATE or TERMINATE the default is MIGRATE (Live Migration).
+    * @param automaticRestart Defines whether the Instance should be automatically
+    *  restarted when it is terminated by Compute Engine (not terminated by user).
+    *  Used when onHostMaintenance is set to TERMINATE.
+    * @param preemptible Defines whether the Instance should be launched as spot instance
+    * @return
+    */
+   @Named("Instances:setScheduling")
+   @POST
+   @Path("/{instance}/setScheduling")
+   @MapBinder(BindToJsonPayload.class)
+   Operation setScheduling(@PathParam("instance") String instanceName,
```

Comment:
@nacx: Thanks for the review comment. Now adding this as an additional overloaded method.

The tests now only exercise the new method (see below). Add another test for the old overloaded version?